docs/DeveloperGuide.md

```diff
@@ -1,34 +1,422 @@
+</span>
+    
 
-## Glossary
+### **Scope**  
+<span style="text-align: justify; display: block">
+This document will cover the structure and software design decisions for the implementation of <b>Nuke</b>. The intended audience for this document are developers, designers and software testers of <b>Nuke</b> <i>or</i> other similar task management application.
+</span>  
 
-* *glossary item* - Definition
+### **Design Goals**  
 
-## Instructions for Manual Testing
+```
+	// To be done.
+```
 
-{Give instructions on how to do a manual product testing e.g., how to load sample data to be used for testing}
+### **Definitions**  
+```
+	// To be done.
+```
+
+<br>
+
+## **Setting Up**  
+Refer to the guide [here](#) to set up.  
+
+<br>
+
+## **Design**  
+
+<br>
+
+## **Implementation**  
+This section will describe the significant details of how certain features in <b>Nuke</b> are being implemented.  
+
+### 1. Add Feature
+#### Overview
+The **add** feature adds modules, categories, tasks and tags into the Module, Category and Task List respectively.
+
+#### 1.1. Add module feature
+
+The add module feature enable the user to add modules into the Module List.
+
+When the user first requests to execute the `addm` command(*assuming the command format given is valid*) to add a module by providing its name, the application will parse the input after `addm`  command word as the module code. From here, there are **three** possible outcomes:
+
+1. The module specified by the user is not provided by NUS -- No module will added.
+2. The module specified by the user is already added -- No module will be added.
+3. The module specified by the user is provided by NUS and have not been added -- Module will be added.
+
+#### Feature Implementation
+
+This feature is facilitated by the `AddModuleCommand` class which add the modules specified by the user. It overrides the `execute()` method which extends from the abstract `AddCommand` class which extends from the abstract `Command` class.   The `execute()` method's role is to execute the adding module operation and do necessary checks .
+
+The `AddCommand` will first try to call the static method `add` in `ModuleManager`  class, which will try to add the module specified by the user, exception will thrown according the following rules:
+
+1. `DuplicateModuleException` will be thrown if the module specified by the user is contained in the `ArrayList` named `moduleList` in `ModuleManager` class.
+2. `ModuleNotProvidedException` will be thrown if the module code specified by the user is not contained in the `HashMap` named `modulesMap` in `ModuleManager` class.
+
+Below are the class-diagram for the involved classes:
+
+![image-20200326014336120](images/Add_Module_Command_Class_Diagram.png)
```

Comment:
The team have clear and useful class diagram, you may add an interface class for Command class.